### PR TITLE
Reimplement bruteforce module

### LINKED ISSUE #7: Reimplement old module: bruteforce
https://github.com/svenstaro/genact/blob/f408ec25fe69eae33a899ee20818b7c10a38b824/genact/modules/bruteforce.py

### PR DESCRIPTION
**Pull request by equal-l2**
Close #7.

https://user-images.githubusercontent.com/8597717/194197032-baa7a722-8267-402e-9b0e-0adc754aa0ed.mp4

### CODE REVIEW

File: src/data.rs [hunk 1]
```diff
@@ -49,3 +49,42 @@ pub static EXTENSIONS_LIST: &[&str] = &[
 
 pub static COMPRESSION_FORMATS_LIST: &[&str] =
     &["gzip", "bzip2", "lzma", "xz", "lzop", "lz4", "zstd"];
+
+pub static PASSWORDS_AND_HASHES_LIST: &[(&str, &str)] = &[
+    (
+        "password",
+        "5e884898da28047151d0e56f8dc6292773603d0d6aabbdd62a11ef721d1542d8",
```
[the remark this answers is not in the document]
Comment (svenstaro) [reply src/data.rs:56]: Also I think we should not really have any passwords here at all. How about you take some random output from [fakers](https://docs.rs/fake/latest/fake/faker/index.html) which we already include? Something like `format!("{}{}", FirstName<EN>, rng.gen_range(10..99))` might give you good enough passwords without including any additional extra data.

File: src/modules/bruteforce.rs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            }
+            newline().await;
+        }
+
+        print(format!("+ Match found -- the password is \"{password}\"")).await;
+        newline().await;
+    }
+}
+
+// color a string rainbow
```
Comment (svenstaro) [inline src/modules/bruteforce.rs:108]: Might it be more maintainable to use [colorful](https://docs.rs/colorful/latest/colorful/trait.Colorful.html) here? We will have to check whether it breaks the web version, though.

File: src/modules/bruteforce.rs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+#[async_trait(?Send)]
+impl Module for Bruteforce {
+    fn name(&self) -> &'static str {
+        "bruteforce"
+    }
+    fn signature(&self) -> String {
+        "./bruteforce.sh".to_string()
+    }
+    async fn run(&self, app_config: &AppConfig) {
```
Comment (svenstaro) [inline src/modules/bruteforce.rs:24]:
```suggestion
impl Module for Bruteforce {
    fn name(&self) -> &'static str {
        "bruteforce"
    }
    
    fn signature(&self) -> String {
        "./bruteforce.sh".to_string()
    }
    
    async fn run(&self, app_config: &AppConfig) {
```

**Review by svenstaro — CHANGES_REQUESTED** [review]
I really like this and it looks awesome! The general logic in the code is good but I think we can clean up the surrounding code a bit and maybe even slim this down a bit.

File: src/modules/bruteforce.rs (hunk 2 above)
Comment (equal-l2) [reply src/modules/bruteforce.rs:110]: I didn't know that crates!
I'll look into it.

Comment (equal-l2) [reply src/modules/bruteforce.rs:110]: Can we use 24-bit colors?
`colorful` seems not to work without 24-bit color support.

Even in 2022 there are a few terminal emulators without the support (macOS terminal, for example)

Comment (equal-l2) [reply src/modules/bruteforce.rs:110]: By the way, can I use 256 colors instead of ANSI 8 colors?
It will make the step of the rainbow colors smoother.

Comment (equal-l2) [reply src/modules/bruteforce.rs]: I found [colorgrad](https://github.com/mazznoer/colorgrad-rs), which generates gradient colors.
I reimplemented the rainbow progress bar with it, along with the smoother colors using 256 colors in 098bfd1.

Screenshot 2022-10-08 at 12 45 23

**Comment by svenstaro** [comment]
Super nice. Very cool module and clean implementation. Thanks!

**Comment by svenstaro** [comment]
How about expanding this a bit and having it crack more than just a single hash?

**Comment by equal-l2** [comment]
Does this match up to what you expected?

https://user-images.githubusercontent.com/8597717/194743698-2ec30d21-b629-4374-b92d-8adc8656c1fc.mov

**Comment by svenstaro** [comment]
Hell yes :D

**Comment by equal-l2** [comment]
Then I'll make a PR with it soon!

**Pull request merged**